### PR TITLE
Refactor/context components

### PR DESCRIPTION
**Pull request by GraceGardner**
What (if any) features are you implementing?
- Add contexts folder and files/components 

What (if anything) did you refactor?
- fixed dbl click bug
- Refactored all of the context and state 

Were there any issues that arose?
- yeah, a lot, but they are fixed now so don't worry about it :)
- issue to think about later is that you can't go to the score card from the menu unless a game has been started, I will add this to the project board 

Is there anything that you need from your teammate?
Please review 

Any other comments, questions, or concerns?
**** FORGOT TO DELETE THE OLD CONTEXT FILE **** 🤦‍♀️

### CODE REVIEW

File: src/Components/ErrorModal.js [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
     <div className='modal'>
       <article className='modal-box'>
         <p>{errorMessage}</p>
         <button className='modal-btn' onClick={() => setErrorMessage({error: null})}>OK</button>
       </article>
     </div>
-  )
+  )} else {
+    return(
+      <></>
```
Comment (philalewis) [inline src/Components/ErrorModal.js:18]: I like the idea of just returning an empty fragment. This is a really cool way of handling this logic.

File: src/Components/Navbar.js [hunk 2]
```diff
@@ -1,22 +1,35 @@
...
 import '../Styles/Navbar.scss'
 
-const Navbar = ({toggleClick}) => {
+const Navbar = () => {
+
+  const [clicked, setClick] = useState(false)
+
+  const toggleClick = () => {
+    console.log(clicked)
+    setClick(!clicked)
```
Comment (philalewis) [inline src/Components/Navbar.js:14]: Very cool way of toggling this

File: src/Contexts/CourseInfoProvider.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    const [ courseInfo, setCourseInfo ] = useState({
+      courses: [],
+      searchURL: '',
+      currentCourse: {}
+    })
+
+    return(
+      <CourseInfo.Provider value={{ courseInfo, setCourseInfo }}>
+        {children}
```
Comment (philalewis) [inline src/Contexts/CourseInfoProvider.js:15]: Good to know that this is a part of React. I did not see that `{children}` is an option in any of the documentation.

File: src/Contexts/LeagueMembersProvider.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import { Errors } from './ErrorsProvider'
+
+export const LeagueMembers = createContext()
+
+export const LeagueMembersProvider = ({ children }) => {
+  const { errorMessage, setErrorMessage } = useContext(Errors)
+
+  const [ leagueMembers, setLeagueMembers ] = useState([])
+
+  useEffect(() => {
```
Comment (philalewis) [inline src/Contexts/LeagueMembersProvider.js:12]: Excellent place to put the useEffect to get all league members.

File: src/Contexts/ScorecardInfoProvider.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React, { createContext, useState } from 'react'
+
+export const ScorecardInfo = createContext()
+
+export const ScorecardInfoProvider = ({ children }) => {
+
+  const [ scorecard, setScorecard ] = useState({
```
Comment (philalewis) [inline src/Contexts/ScorecardInfoProvider.js:7]: We should consider splitting this up in the future. This might be too much to keep track of in one spot.

**Review by philalewis — APPROVED** [review]

File: src/Components/Navbar.js [hunk 6]
```diff
@@ -1,22 +1,35 @@
...
+    <>
+      <nav>
       <img
         className='logo'
         src={logo}
         alt='Discify Logo'
       />
-      <button className='dropdown-button' onClick={() => toggleClick()}>
-        <FontAwesomeIcon icon={ faBars }/>
+      <button
```
Comment (oacortez) [inline src/Components/Navbar.js:25]: Thank you for adjusting this button component and attributes much easier to read

**Review by oacortez — APPROVED** [review]
Thank you for making this branch grace! I see a lot of refactoring of our global state organization. I like how we divided more of our global state into component, which in there you are creating context and holding default keys and values. Also, nice job fixing the double click bug we had in the past branches! I will merge this branch.

**Pull request merged**